### PR TITLE
Issue#315-rate-limit-support

### PR DESCRIPTION
**Pull request by LuitelSamikshya**

### CODE REVIEW

File: runner/runner.go [hunk 1]
```diff
@@ -426,10 +436,12 @@ func (r *Runner) RunEnumeration() {
 			for _, p := range r.options.requestURIs {
 				scanopts := r.scanopts.Clone()
 				scanopts.RequestURI = p
+				r.ratelimiter.Take()
```
Comment (Mzack9999) [inline runner/runner.go:439]: I think we can remove it here as it's already called in the `r.process` function (line 467)

File: runner/runner.go [hunk 2]
```diff
@@ -426,10 +436,12 @@ func (r *Runner) RunEnumeration() {
...
 				r.process(t, &wg, r.hp, protocol, scanopts, output)
 				reqs++
 			}
 		} else {
+			r.ratelimiter.Take()
```
Comment (Mzack9999) [inline runner/runner.go:444]: I think we can remove it here as it's already called in the `r.process` function (line 467)

**Review by Mzack9999 — CHANGES_REQUESTED** [review]

File: runner/runner.go (hunk 1 above)
Comment (LuitelSamikshya) [reply runner/runner.go:439]: I ran some  tests removing this line but RPS seems to exceed the rate limit.

File: runner/runner.go (hunk 2 above)
Comment (LuitelSamikshya) [reply runner/runner.go:444]: I ran some  tests removing this line but RPS seems to exceed the rate limit.

**Review by Mzack9999 — APPROVED** [review]

**Pull request merged**